src/app/features/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -0,0 +1,62 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+import { StatusCodes, TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { PsvTechRecord } from '../../forms/templates/psv/psv-tech-record.template';
+import { HgvTechRecord } from '../../forms/templates/hgv/hgv-tech-record.template';
+import { TrlTechRecord } from '../../forms/templates/trl/trl-tech-record.template';
+import { FormNode } from '../../forms/services/dynamic-form.service';
+
+@Component({
+  selector: 'app-tech-record-summary',
+  templateUrl: './tech-record-summary.component.html',
+  styleUrls: ['./tech-record-summary.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TechRecordSummaryComponent implements OnInit {
+  @Input() vehicleTechRecord!: VehicleTechRecordModel;
+  template!: FormNode;
+  currentRecord!: TechRecordModel | undefined;
+
+  ngOnInit(): void {
+    this.vehicleTemplate();
+    this.currentRecord = this.viewableTechRecord(this.vehicleTechRecord)
+  }
+
+  constructor() {
+  }
+
+  viewableTechRecord(record: VehicleTechRecordModel): TechRecordModel | undefined {
+    let viewableTechRecord = record?.techRecord?.find(record => record.statusCode === StatusCodes.PROVISIONAL)
+    if (viewableTechRecord == undefined){
+      viewableTechRecord = record?.techRecord?.find(record => record.statusCode === StatusCodes.CURRENT)
+    }
+    if (viewableTechRecord == undefined){
+      viewableTechRecord = record?.techRecord?.find(record => record.statusCode === StatusCodes.ARCHIVED)
+    }
```

Comment:
Regardless of the conversation above. Why are both of these checking for undefined? if `viewableTechRecord` is undefined then the second condition will overwrite the first

Reply:
Yeah that's what we want. If it is not assigned when we find on provisional it will be undefined, so try current, if that is also unassigned try archived. Means if it finds it in current it won't be placed by archive.

Reply:
Could have a single find and use the `OR` operator.